convert2rhel/repo.py

```diff
@@ -47,3 +54,57 @@ def get_rhel_repoids():
     loggerinst.info("RHEL repository IDs to enable: %s" % ", ".join(repos_needed))
 
     return repos_needed
+
+
+def download_repofile(repofile_url):
+    """Download a repository file from a specific URL.
+
+    :raises exceptions.CriticalError: When the repository file URL is inaccessible.
+    :returns str: Contents of the repofile if successfully downloaded.
+    """
+    try:
+        with closing(urllib.request.urlopen(repofile_url, timeout=15)) as response:
+            contents = response.read()
+
+            if not contents:
+                loggerinst.critical_no_exit("")
+                raise exceptions.CriticalError(
+                    id_="REPOSITORY_FILE_EMPTY_CONTENT",
+                    title="No content was availble in requested repository file.",
+                    description="The requested repository file seems to be empty. No content received when checking for url: %s"
+                    % repofile_url,
+                )
+
+            loggerinst.info("Successfully downloaded the requested repofile from %s." % repofile_url)
+            return contents.decode()
+    except urllib.error.URLError as err:
+        raise exceptions.CriticalError(
+            id_="DOWNLOAD_REPOSITORY_FILE_FAILED",
+            title="Failed to download repository file.",
```

Comment:
```suggestion
            title="Failed to download repository file",
```
We don't use full stops in titles.